substratools/algo.py

```diff
@@ -182,6 +182,11 @@ def __init__(self, interface, workspace=None, opener_wrapper=None):
             opener.load_from_module(workspace=self._workspace)
         self._interface = interface
 
+    def _assert_output_model_exists(self):
+        path = self._workspace.output_model_path
+        assert not os.path.isdir(path), f'Expected output model file at {path}, found dir'
+        assert os.path.isfile(path), f'Output model file {path} does not exists'
```

Comment:
In my opinion it would be simpler to check that the path exists and then that the path is a file. It's a bit misleading here, the message and the test don't match.

Reply:
`os.path.isfile` and `os.path.isdir` return false if the path doesn't exist. So technically, the `isfile` check is the only one that is necessary.

I added `isdir` check because I noticed that ML libraries sometimes save a given object at `path/defaultname.customformat` when given `path` and you have to write additional code to save it as just `path`. I thought adding the `isdir` check would help the user during debugging.

Reply:
Yes I think it's a very good idea to have the 2 different exceptions. It will as you said greatly help debugging.
I think it will be more natural to check first if the path exists and then check the path points to a file. 

This is really minor and my point of view so we can keep it like this :)